frontend/src/horses/AddHorseModal.tsx

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios';
+import React, {useState} from 'react';
+import Modal from 'react-modal';
+import {HorseModel} from "./HorseModel";
+import FieldLabelGroup from "../structuralComponents/FieldLabelGroup";
+import CloseIcon from "../icons/CloseIcon";
+import "../index/css/AddItemModal.css";
+import "../buttons/css/SubmitButton.css";
+
+type ModalProps = {
+    modalIsOpen: boolean,
+    closeModal: () => void,
+    reloadHorses: () => void,
+    setSuccessMessage: (input: string) => void,
+}
+
+function AddItemModal(props: ModalProps) {
+    const [newHorse, setNewHorse] = useState<HorseModel>({
+        id: "", name: "", owner: "", consumption: []
+    })
+
+
+    const saveNewHorse = () => {
+        axios.post("/horses/", newHorse)
+            .catch((e) => console.error("POST Error: " + e))
+            .then(props.reloadHorses)
+            .then(props.closeModal)
+            .then(() => setNewHorse({id: "", name: "", owner: "", consumption: []}))
+            .then(() => props.setSuccessMessage("Eintrag erfolgreich hinzugefügt"))
+
+    }
+
+    const handleSubmit = (event: React.SyntheticEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        saveNewHorse()
+    }
+    const handleChange = (event: any) => {
+        const name = event.target.name;
+        const value = event.target.value;
```

Comment:
Not an issue but use Destructuring as  shortcut :
const  {name, value} = event.target

Reply:
Didn't know about that, cool!